src/rust/bitbox02-rust/src/bb02_async.rs

```diff
@@ -48,8 +48,8 @@ impl<O> core::future::Future for AsyncOption<'_, O> {
 
 /// Waits for an option to contain a value and returns that value, leaving `None` in its place.
 /// E.g. `assert_eq!(option(&Some(42)).await, 42)`.
-pub fn option<'a, O>(option: &'a RefCell<Option<O>>) -> AsyncOption<'a, O> {
-    AsyncOption(&option)
+pub fn option<O>(option: &RefCell<Option<O>>) -> AsyncOption<'_, O> {
+    AsyncOption(option)
```

Comment:
just checking: this was not clippy, right.

Reply:
I guess not. I really don't like that double reference somehow just works. The previous version should have been a compilation error imho.

Reply:
you are not alone, however auto-deref makes the language much more ergonomic: https://stackoverflow.com/questions/28519997/what-are-rusts-exact-auto-dereferencing-rules